packages/samples/headless-ssr/cypress/e2e/ssr.cy.ts

```diff
@@ -0,0 +1,56 @@
+import 'cypress-web-vitals';
+
+describe('headless ssr example', () => {
+  const numResults = 10;
+  const numResultsMsg = `Hydrated engine with ${numResults} results`;
+  const msgSelector = '#hydrated-msg';
+  const timestampSelector = '#timestamp';
+  it('renders page in SSR as expected', () => {
+    cy.intercept('/', (req) => {
+      req.continue((resp) => {
+        const dom = new DOMParser().parseFromString(resp.body, 'text/html');
+        expect(dom.querySelector(msgSelector)?.textContent).to.equal(
+          numResultsMsg
+        );
+        expect(dom.querySelectorAll('li').length).to.equal(numResults);
+      });
+    });
+    cy.visit('/');
+  });
+
+  it('renders page in CSR as expected', () => {
+    cy.visit('/');
+    cy.get(msgSelector).should('have.text', numResultsMsg);
+    cy.get('li').should('have.length', numResults);
+  });
+
+  it('renders result list in SSR and then in CSR', () => {
+    const interceptAlias = 'searchResults';
+    cy.intercept('/').as(interceptAlias);
+
+    cy.visit('/');
+    cy.wait(`@${interceptAlias}`).then((intercept) => {
```

Comment:
You should be able to use an async/await in the `it` function.

Reply:
`cy.wait` returns a `Chainable<Interception>` and not a `Promise`. Using `await` results in the error: `Type of 'await' operand must either be a valid promise or must not contain a callable 'then' member.`